pkg/solana/logpoller/parser_test.go

```diff
@@ -0,0 +1,278 @@
+			" WHERE chain_id = :solana_chain_id " +
+				"ORDER BY block_number ASC, block_timestamp DESC")
+
+		require.NoError(t, err)
+		assert.Equal(t, expected, result)
+
+		assertArgs(t, args, 1)
+	})
+
+	t.Run("basic query with default primitives no order by and cursor", func(t *testing.T) {
+		t.Parallel()
+
+		parser := &pgDSLParser{}
+		expressions := []query.Expression{
+			query.Timestamp(10, primitives.Eq),
+			query.TxHash(txHash),
+			query.Block("99", primitives.Neq),
+			query.Confidence(primitives.Finalized),
+		}
+		limiter := query.NewLimitAndSort(query.CursorLimit(fmt.Sprintf("10-20-%s", txHash), query.CursorPrevious, 20))
+
+		result, args, err := parser.buildQuery(chainID, expressions, limiter)
+		expected := logsQuery(
+			" WHERE chain_id = :solana_chain_id " +
+				"AND (block_timestamp = :block_timestamp_0 AND tx_hash = :tx_hash_0 " +
+				"AND block_number != :block_number_0) " +
+				"AND (block_number < :cursor_block_number OR (block_number = :cursor_block_number " +
+				"AND log_index < :cursor_log_index)) " +
+				"ORDER BY block_number DESC, log_index DESC, tx_hash DESC LIMIT 20")
+
+		require.NoError(t, err)
+		assert.Equal(t, expected, result)
+
+		assertArgs(t, args, 6)
+	})
+
+	t.Run("query for finality", func(t *testing.T) {
+		t.Parallel()
+
+		t.Run("finalized", func(t *testing.T) {
+			parser := &pgDSLParser{}
+			expressions := []query.Expression{query.Confidence(primitives.Finalized)}
+			limiter := query.LimitAndSort{}
+
+			result, args, err := parser.buildQuery(chainID, expressions, limiter)
+			expected := logsQuery(
+				" WHERE chain_id = :solana_chain_id " +
+					"ORDER BY " + defaultSort)
+
+			require.NoError(t, err)
+			assert.Equal(t, expected, result)
+
+			assertArgs(t, args, 1)
+		})
+
+		t.Run("unconfirmed", func(t *testing.T) {
+			parser := &pgDSLParser{}
+			expressions := []query.Expression{query.Confidence(primitives.Unconfirmed)}
+			limiter := query.LimitAndSort{}
+
+			result, args, err := parser.buildQuery(chainID, expressions, limiter)
+			expected := logsQuery(
+				" WHERE chain_id = :solana_chain_id " +
+					"ORDER BY " + defaultSort)
+
+			require.NoError(t, err)
+			assert.Equal(t, expected, result)
+
+			assertArgs(t, args, 1)
+		})
+	})
+
+	// nested query -> a & (b || c)
+	t.Run("nested query", func(t *testing.T) {
+		t.Parallel()
+
+		parser := &pgDSLParser{}
+
+		expressions := []query.Expression{
+			{BoolExpression: query.BoolExpression{
+				Expressions: []query.Expression{
+					query.Timestamp(10, primitives.Gte),
+					{BoolExpression: query.BoolExpression{
+						Expressions: []query.Expression{
+							query.TxHash(txHash),
+							query.Confidence(primitives.Unconfirmed),
+						},
+						BoolOperator: query.OR,
+					}},
+				},
+				BoolOperator: query.AND,
+			}},
+		}
+		limiter := query.LimitAndSort{}
+
+		result, args, err := parser.buildQuery(chainID, expressions, limiter)
+		expected := logsQuery(
+			" WHERE chain_id = :solana_chain_id " +
```

Comment:
I think this should just be `:chain_id` rather than `:solana_chain_id`, to match the db schema.

Also, hoping we can move this code to `chainlink-framework` at some point to re-use it for many chain families, abstracting it a bit to fill in the custom data types.